src/OpenLoco/src/Jukebox.cpp

```diff
@@ -44,98 +47,90 @@ namespace OpenLoco::Jukebox
         { PathId::music_sandy_track_blues, StringIds::music_sandy_track_blues, 1921, 1929 }
     };
 
-    static std::vector<MusicId> makeAllMusicPlaylist()
+    // Queue that stores all the songs in the current playlist
+    static std::queue<MusicId> playlist;
+
+    static std::vector<MusicId> findAllSongs()
     {
-        std::vector<MusicId> playlist(kNumMusicTracks);
-        std::iota(playlist.begin(), playlist.end(), 0);
-        return playlist;
+        std::vector<MusicId> songs(kNumMusicTracks);
+        std::iota(songs.begin(), songs.end(), 0);
+        return songs;
     }
 
-    static std::vector<MusicId> makeCurrentEraPlaylist()
+    static std::vector<MusicId> findCurrentEraSongs()
     {
-        auto playlist = std::vector<MusicId>();
+        std::vector<MusicId> songs;
         auto currentYear = getCurrentYear();
 
         for (auto i = 0; i < kNumMusicTracks; i++)
         {
             const auto& mi = kMusicInfo[i];
             if (currentYear >= mi.startYear && currentYear <= mi.endYear)
             {
-                playlist.push_back(i);
+                songs.push_back(i);
             }
         }
 
-        return playlist;
+        return songs;
     }
 
-    static std::vector<MusicId> makeCustomSelectionPlaylist()
+    static std::vector<MusicId> findCustomSelectionSongs()
     {
-        auto playlist = std::vector<MusicId>();
+        std::vector<MusicId> songs;
```

Comment:
if you use coroutines here (specifically `co_yield`) you won't need this temporary

Reply:
So I should leave it as auto?